datafusion/common/src/scalar.rs

```diff
@@ -2872,6 +2872,14 @@ impl TryFrom<&DataType> for ScalarValue {
                 ))),
                 1,
             )),
+            DataType::Map(field, _) => ScalarValue::List(new_null_array(
```

Comment:
I think the data type needs to match the `ScalarValue` enum (e.g. we need to add `DataType::Map` to make this work properly). I'll keep the discussion going at https://github.com/apache/arrow-datafusion/issues/8262#issuecomment-1852700799